src/server.rs

```diff
@@ -85,7 +85,15 @@ pub async fn run(
     // with rustls, the hyper and tower crates [what axum is built on] had to
     // be used directly
 
-    let config = OrchestratorConfig::load(config_path).await;
+    let config = OrchestratorConfig::load(config_path)
+        .await
+        .map_err(Error::from)
+        .map_err(|error| match error {
+            Error::ConfigurationFailed(_) => error,
+            _ => {
+                panic!("Unexpected error during service configuration: {error}");
+            }
+        })?;
```

Comment:
Since we are `panic!`ing for config errors, why not just `panic!` from within `Orchestrator.load()` instead of returning an `Error`, mapping it to `server::Error`, to ultimately discard it? I'm not against it, just not seeing the benefit of using `Error` for this as we don't do anything with it.

`server::Error` is specifically for returning error responses to the client from route handlers.

Reply:
I promise this is the last remaining artifact from the `panic` vs `fatal` discussion, if the below explanation doesn't not seem valuable I will rescind on this and make all configuration errors just panic, but let me give it one last shot :)

> `server::Error` is specifically for returning error responses to the client from route handlers.

I did gather this was the intention of this error type, but you will notice that regardless of this PR, `Server::run()` also returns a `server::Error`. In this PR you'll see that nowhere else in `Server::run()` is an error actually returned, all `server::Error`s are handled. 

_As I am typing this I am thinking that some revision should be made to what happens here regardless, so for now lets imagine that instead `server::run()` will handle any `server::Error`s and instead returns either a `config::Error` or generic `std::error::Error` (so imagine no more `map_err`s in the above)._

> Since we are panic!ing for config errors,

And this is where the distinction between a graceful exit vs unexpected exit comes up. Regarding errors, if the developer *really doesn't expect* that a result will be an error, they should `panic!()/.expect()`. However when the developer knows how something can fail and needs to *check success before continuing*, this is the typical error handling flow. When an error can be expected, the module of code should check for an error and then either handle the error if it can, or propagate it down. This behaviour can be consistent all the way back up to main where the final error may cause failure if it can not be handled. (If the system is complex and managing exit points is difficult, this is where having something like `report_fatal(error) { error!(error); std::process::exit(error.code.unwrap_or(1)); }` comes up, but this codebase does not have this problem.

Note about
```
map_err(|error| match error {
            Error::ConfigurationFailed(_) => error,
            _ => {
                panic!("Unexpected error during service configuration: {error}");
            }
        })?;
```

_Just to avoid any confusion, because of the (unnecessary, to-be-removed) error mapping to `server::Error`, this `panic!` existed to properly notify a developer of unexpected behaviour during the error handling, this panic is not the panic for a failed configuration, it exists to catch unexpected behaviour caused by developers misusing code._

**So this is the intention of the above code:** Config validation failure is an error that can not be handled at any point in execution, but can be expected in a way that we can fully report the source of the error without needing to send a developer in to investigate an unknown source of the problem in code and then debug it. So for the sole reason of keeping application development and application usage as separate flows, I believe that configuration failure should cause the application to terminate with a non-zero exit code as a result of typical error handling, and we can do this without introducing alternate exit points by propagating up to main, instead of terminating in a way designed for atypical or unexpected scenarios. If this distinction has any merit and seems valid let me know, otherwise I have no issues and can just panic.